models/classes/runner/time/QtiTimeConstraint.php

```diff
@@ -158,8 +151,17 @@ protected function getRemainingTimeFrom(QtiDuration $duration)
      */
     public function getMaximumRemainingTime()
     {
-        if (($timeLimits = $this->getSource()->getTimeLimits()) !== null && ($maxTime = $timeLimits->getMaxTime()) !== null) {
-            return $this->getRemainingTimeFrom($maxTime);
+        if (($maxTime = $this->getAdjustedMaxTime()) !== null) {
+            $maximumTime = clone $maxTime;
+
+            if ($this->getApplyExtraTime() && $this->timer) {
```

Comment:
Is it valid case to apply extra time on top of timer adjustments from TimerAdjustmentMap?

Reply:
Although it is not possible to enable both "extra time" and "time adjustment" functionality for proctor, but the current `extended_time` LTI parameter implementation is employing the "extra time".
We do plan to reimplement `extended_time` with timer adjustments, but until then I've tried to make it work.